x-pack/plugins/index_management/public/application/sections/home/data_stream_list/edit_data_retention_modal/edit_data_retention_modal.tsx

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiButtonEmpty,
+  EuiButton,
+  EuiSpacer,
+  EuiCallOut,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  useForm,
+  useFormData,
+  Form,
+  fieldFormatters,
+  FormSchema,
+  FIELD_TYPES,
+  UseField,
+  ToggleField,
+  NumericField,
+} from '../../../../../shared_imports';
+
+import { splitSizeAndUnits } from '../../../../../../common';
+import { useAppContext } from '../../../../app_context';
+import { UnitField } from './unit_field';
+import { updateDataRetention } from '../../../../services/api';
+
+interface Props {
+  dataRetention: string;
+  dataStreamName: string;
+  configuredByILM: boolean;
+  onClose: (data?: { hasUpdatedDataRetention: boolean }) => void;
+}
+
+const DEFAULT_DATA_RETENTION_PERIOD = '7d';
```

Comment:
I'm defaulting in the UI  to `7d` retention in case there arent any values set in the dataRetention field